### PR TITLE
change add import vitals to observation form

### PR DESCRIPTION
**Pull request by sjpadgett**
add options to exportcat1service from previous closed PR

We now write physical exams vitals to observations and vitals

### CODE REVIEW

**Comment by sjpadgett** [comment]
Companioned with https://github.com/openemr/openemr/pull/5119

**Comment by sjpadgett** [comment]
getting cypress {"qrda_export_1649730489\\CMS22v10\\1_Alex_Pittman.xml"=>"undefined method `_type' for nil:NilClass"} on every export now.

**Comment by sjpadgett** [comment]
image

**Comment by sjpadgett** [comment]
@kchapple 
Darned dates! Example measure calculation for an exclusion to show the importance of carrying time around charted data:

image

@bradymiller I know you've wrestled with dates going way back. Please keep an eye on what I'm doing for any gotchas I may get myself into... I've got encounters covered however, I still have to ensure only date is looked at when searching for encounters to populate items associated with encounters e.g. problems etc.. a datetime may not match.

**Comment by sjpadgett** [comment]
looks like between list table and building report, time is changed
```
<effectiveTime><low value='20200327123000'/><high nullFlavor='UNK'/></effectiveTime>
                    <!-- QDM Attribute: Code -->
       <value xsi:type="CD" code="I10" codeSystem="2.16.840.1.113883.6.90" codeSystemName="ICD10"/>
```
table:

image

File: src/Services/Qrda/Util/DateHelper.php [hunk 1]
```diff
@@ -20,7 +20,7 @@ class DateHelper
      */
     public static function format_datetime_cqm($datetime)
     {
-        return gmdate('Y-m-d\TH:i:s.u', date('U', strtotime($datetime)));
+        return !empty($datetime) ? date('Y-m-d\TH:i:s.u', strtotime($datetime)) : null;
```
Comment (sjpadgett) [inline src/Services/Qrda/Util/DateHelper.php:23]: @kchapple you may want to do this differently but it's important we keep local same as the imported datetimes.
I get to bumping the cqm execution module dependency tomorrow...

**Comment by kchapple** [comment]
Thanks @sjpadgett looks good. Thanks!

I don't know if these matter for testing, but there are some small holes I've noticed with dates. 

1. For medication order careplan, the model takes a start datetime, but we only have date.
2. For form_observations, we don't have end datetime.
3. For encounters, we're not populating events table (where I would get duration) so we don't have a real end-datetime so I'm fudging it

**Comment by kchapple** [comment]
@sjpadgett I may be misunderstanding the issue, I don't see an update to SQL to change lists table begdate and enddate to datetime types. It looks like that's what you had in your screenshot above, maybe you forgot to check int sql?

**Comment by sjpadgett** [comment]
@kchapple yeah forgot file. I'll bring this in soon as actions okays...

**Comment by kchapple** [comment]
@sjpadgett probably need database.sql also

**Comment by kchapple** [comment]
@sjpadgett Here's a small commit to add end date to care plans
https://github.com/mi-squared/openemr/commit/f3a70125720740e9a977afb8d2477151b91fde84

**Comment by sjpadgett** [comment]
@kchapple I'm moving a little slow today. Was there something I needed to do with medication import?

**Comment by kchapple** [comment]
@sjpadgett Just that the 'date' column of form_care_plan that we are using for the "low" datetime on the QDM models is a date type, not a datetime. I don't know if it matters.

**Comment by sjpadgett** [comment]
@kchapple Will you please check me on my database upgrades. I'm making silly mistakes this morning!:)

**Comment by kchapple** [comment]
@sjpadgett The updates to dates look good. The only thing I noticed in testing just now was that the enddate for medical_problems (maybe other issues too?) in the 'lists' are being imported as 0000-00-00 00:00:00 if they are blank, rather than NULL, and this is messing up the OpenEMR UI to display the issue as inactive. Is there a way to set the datetimes to NULL if there is no valid value?

**Comment by kchapple** [comment]
@sjpadgett The other thing are the UI controls on Observation and Care Plan forms.... they are datepickers, not datetime pickers, so there's no way to put in a datetime, or edit an existing datetime. Maybe I'm nitpicking, LOL.

**Comment by kchapple** [comment]
I have to go pick up kids, be back in an hour or two.

File: sql/6_1_0-to-7_0_0_upgrade.sql [hunk 2]
```diff
@@ -290,3 +290,24 @@ ALTER TABLE `insurance_companies` CHANGE `inactive` `inactive` TINYINT(1) NOT NU
 UPDATE `layout_options` SET `fld_length` = '8' WHERE `layout_options`.`form_id` = 'DEM' AND `layout_options`.`field_id` = 'postal_code';
 #EndIf
```
Comment (bradymiller) [inline sql/6_1_0-to-7_0_0_upgrade.sql:291]: @sjpadgett , There are some discrepancies here between the upgrade and database.sql. I plan to fix these so you can keep focusing on cqm :)

File: sql/6_1_0-to-7_0_0_upgrade.sql [hunk 3]
```diff
@@ -290,3 +290,24 @@ ALTER TABLE `insurance_companies` CHANGE `inactive` `inactive` TINYINT(1) NOT NU
...
+ALTER TABLE `form_observation` CHANGE `ob_code` `ob_code` VARCHAR(64) NULL DEFAULT NULL, CHANGE `ob_type` `ob_type` VARCHAR(64) NULL DEFAULT NULL, CHANGE `ob_reason_code` `ob_reason_code` VARCHAR(64) NULL DEFAULT NULL;
+#EndIf
+
+#IfMissingColumn form_care_plan reason_status
+ALTER TABLE `form_care_plan` ADD `reason_status` VARCHAR(31) NULL DEFAULT NULL;
+#EndIf
+
+#IfNotColumnType lists begdate datetime
+ALTER TABLE `lists` CHANGE `begdate` `begdate` DATETIME NULL DEFAULT NULL;
+ALTER TABLE `lists` CHANGE `enddate` `enddate` DATETIME NULL DEFAULT NULL;
```
Comment (bradymiller) [inline sql/6_1_0-to-7_0_0_upgrade.sql:304]: add these changes in lists to database.sql

File: sql/6_1_0-to-7_0_0_upgrade.sql [hunk 4]
```diff
@@ -290,3 +290,24 @@ ALTER TABLE `insurance_companies` CHANGE `inactive` `inactive` TINYINT(1) NOT NU
 UPDATE `layout_options` SET `fld_length` = '8' WHERE `layout_options`.`form_id` = 'DEM' AND `layout_options`.`field_id` = 'postal_code';
 #EndIf
 
+#IfNotColumnType form_observation date datetime
```
Comment (bradymiller) [inline sql/6_1_0-to-7_0_0_upgrade.sql:293]: in database.sql date is already datetime, so this block will be skipped
will separate this out into separate if blocks
also need to add new date_end field here

Comment (bradymiller) [reply sql/6_1_0-to-7_0_0_upgrade.sql:293]: disregard above comment, i messed up. this block looks good :)

File: sql/6_1_0-to-7_0_0_upgrade.sql (hunk 3 above)
Comment (bradymiller) [reply sql/6_1_0-to-7_0_0_upgrade.sql:304]: done, the only issue was missing these changes in lists in database.sql. just fixed and pushed to master. go cqm!

Comment (sjpadgett) [reply sql/6_1_0-to-7_0_0_upgrade.sql:304]: Thanks Brady ole eagle eye,

**Pull request merged**